File: XamarinStore.iOS/Views/TextEntryView.cs
using System;
using MonoTouch.UIKit;
using System.Drawing;

namespace XamarinStore
{
	public class TextEntryView : UIView
	{
		public Action<string> ValueChanged = (s)=> {};
		readonly protected UITextField textField;
		public TextEntryView () : base (new RectangleF(0,0,320,44))
		{
			AddSubview (textField = new UITextField {
				BorderStyle = UITextBorderStyle.RoundedRect,
				ShouldReturn = (tf)=>{
					tf.ResignFirstResponder();
					return true;
				},
			});
		}
		public UIKeyboardType KeyboardType
		{
			get{ return textField.KeyboardType; }
			set{ textField.KeyboardType = value; }
		}

		public UITextAutocapitalizationType AutocapitalizationType
		{
			get { return textField.AutocapitalizationType; }
			set { textField.AutocapitalizationType = value; }
		}

		public string PlaceHolder {
			get {
				return textField.Placeholder;
			}
			set {
				textField.Placeholder = value;
			}
		}

		public string Value {
			get {
				return textField.Text;
			}
			set {
				textField.Text = value;
				ValueChanged (textField.Text);
			}
		}

		public override void LayoutSubviews ()
		{
			const float sidePadding = 10f;
			const float topPadding = 5f;
			base.LayoutSubviews ();

			var width = (Bounds.Width - (sidePadding * 2));
			var height = Bounds.Height - (topPadding * 2);

			textField.Frame = new RectangleF (sidePadding, topPadding, width, height);
		}
	}
}

File: F#/XamarinStore.iOS/Views/TextEntryView.fs
namespace XamarinStore

open System
open MonoTouch.UIKit
open System.Drawing

type TextEntryView() as this =
    inherit UIView(new RectangleF(0.0f, 0.0f, 320.0f, 44.0f))

    let textField = new UITextField ( BorderStyle = UITextBorderStyle.RoundedRect,
                                      ShouldReturn = fun tf -> tf.ResignFirstResponder() |> ignore
                                                               true)
    do this.AddSubview textField

    member val TextField = textField with get,set

    member val ValueChanged = fun (x:string) -> () with get,set

    member this.KeyboardType
        with get () = textField.KeyboardType
        and set value = textField.KeyboardType <- value

    member this.AutocapitalizationType
        with get () = textField.AutocapitalizationType
        and set value = textField.AutocapitalizationType <- value

    member this.PlaceHolder
        with get () = textField.Placeholder
        and set value = textField.Placeholder <- value

    member this.Value
        with get () = textField.Text
        and set value = textField.Text <- value
                        this.ValueChanged (textField.Text)

    override this.LayoutSubviews () =
        let sidePadding = 10.0f
        let topPadding = 5.0f
        base.LayoutSubviews ()

        let width = this.Bounds.Width - sidePadding * 2.0f
        let height = this.Bounds.Height - topPadding * 2.0f

        textField.Frame <- new RectangleF (sidePadding, topPadding, width, height)
